spec/models/status_spec.rb

```diff
@@ -317,38 +317,43 @@
       @followed_public_status = Fabricate(:status, account: followed, visibility: :public)
       @followed_direct_status = Fabricate(:status, account: followed, visibility: :direct)
       @not_followed_direct_status = Fabricate(:status, account: not_followed, visibility: :direct)
-
-      @results = Status.as_direct_timeline(account)
     end
 
     it 'does not include public statuses from self' do
-      expect(@results).to_not include(@self_public_status)
+      results = Status.as_direct_timeline(account)
+      expect(results).to_not include(@self_public_status)
     end
 
     it 'includes direct statuses from self' do
-      expect(@results).to include(@self_direct_status)
+      results = Status.as_direct_timeline(account)
+      expect(results).to include(@self_direct_status)
     end
 
     it 'does not include public statuses from followed' do
-      expect(@results).to_not include(@followed_public_status)
+      results = Status.as_direct_timeline(account)
+      expect(results).to_not include(@followed_public_status)
     end
 
     it 'includes direct statuses mentioning recipient from followed' do
       Fabricate(:mention, account: account, status: @followed_direct_status)
-      expect(@results).to include(@followed_direct_status)
+      results = Status.as_direct_timeline(account)
+      expect(results).to include(@followed_direct_status)
     end
 
     it 'does not include direct statuses not mentioning recipient from followed' do
-      expect(@results).to_not include(@followed_direct_status)
+      results = Status.as_direct_timeline(account)
+      expect(results).to_not include(@followed_direct_status)
     end
 
     it 'includes direct statuses mentioning recipient from non-followed' do
       Fabricate(:mention, account: account, status: @not_followed_direct_status)
-      expect(@results).to include(@not_followed_direct_status)
```

Comment:
`@results` will be created in the before block.
and Mention object is created after the before block is executed.